store.go

```diff
@@ -118,3 +177,130 @@ func (b *BucketReaderAt) ReadAt(p []byte, off int64) (n int, err error) {
 
 	return rc.Read(p)
 }
+
+// ErrNonTimestampCol is returned by compareTimestamp when the expression is not for the given timestamp column.
+var ErrNonTimestampCol = errors.New("non timestamp column")
+
+// compareTimestamp returns true if the block contains timestamps covered in expr.
+func compareTimestamp(col string, start, end uint64, expr logicalplan.Expr) (bool, error) {
+	if expr == nil {
+		return false, nil
+	}
+
+	parseExpr := func(e *logicalplan.BinaryExpr) (uint64, error) {
```

Comment:
I feel like having this expression evaluation code feels wrong in `store.go`. Is there some way to extract this and the switch below to into some sort of `expr.Eval` function to be reused?